api/api/serializers/media_serializers.py

```diff
@@ -628,37 +684,56 @@ class MediaSearchRequestSourceSerializer(serializers.Serializer):
         Keep the fields names in sync with the actual fields below as this list is
         used to generate Swagger documentation.
         """
-
-        _field_attrs = {
-            "help_text": (
-                "A comma separated list of data sources; valid values are "
-                "``source_name``s from the stats endpoint: "
-                f"{settings.CANONICAL_ORIGIN}/v1/{media_path}/stats/."
-            ),
-            "required": False,
+        source_help_text = SOURCE_HELP_TEXT % {
+            "origin": settings.CANONICAL_ORIGIN,
+            "media_path": media_path,
+        }
+        excluded_help_text = EXCLUDED_SOURCE_HELP_TEXT % {
+            "origin": settings.CANONICAL_ORIGIN,
+            "media_path": media_path,
         }
 
         source = serializers.CharField(
             label="provider",
-            **_field_attrs,
+            help_text=source_help_text,
+            required=False,
         )
         excluded_source = serializers.CharField(
             label="excluded_provider",
-            **_field_attrs,
+            help_text=excluded_help_text,
+            required=False,
         )
 
-        @staticmethod
-        def validate_source_field(value):
-            """Check whether source is a valid source."""
-
+        def validate_source(self, value):
+            """
+            For the regular searches, split the value.lower() by comma and only return the
+            source names that are in the search controller's sources list for the media type.
+            For the collection=tag, return the value as is. It is ignored in the query builder.
+            For source and creator collections, accept the value as is, without lower-casing
+            or splitting, and check if it's in the valid source name list.
+            This function validates the source and excluded_source fields, but `excluded_source`
+            is ignored for collection requests.
+            """
             allowed_sources = list(search_controller.get_sources(media_type).keys())
-            sources = value.lower().split(",")
-            sources = [source for source in sources if source in allowed_sources]
-            value = ",".join(sources)
-            return value
-
-        def validate_source(self, input_sources):
-            return self.validate_source_field(input_sources)
+            sources_list = ", ".join([f"'{s}'" for s in allowed_sources])
```

Comment:
Nit: this operation is unnecessary for most queries. Can we move it into the error branch itself on line 726 and bypass it for all other search queries?